rumqttd/src/protocol/v5/subscribe.rs

```diff
@@ -140,6 +140,20 @@ mod filter {
         let mut options = 0;
         options |= filter.qos as u8;
 
+        if filter.nolocal {
```

Comment:
This might be unnecessary since the broker isn't expected to write a subscribe onto network if I am not wrong...

Reply:
True, but `protocol` module doesn't make assumption about where it is being used, in future we can extract it out to its own crate like we had `mqttbytes` previously, so its good to keep it updated